### PR TITLE
Create a TestStepResult for each test report (without BG worker)

### PR DESCRIPTION
**Pull request by gaborszakacs**

### CODE REVIEW

File: bitrise/bitrise.go [hunk 1]
```diff
@@ -27,13 +29,46 @@ type Client struct {
...
+	AnnotationLevel string `json:"annotation_level"`
+	Message         string `json:"message"`
+	Title           string `json:"title"`
+	RawDetails      string `json:"raw_details"`
+}
+
 // NewClient returns a new instance of *Client.
 func NewClient(apiToken string) *Client {
-	return &Client{
+	cl := Client{
```
Comment (zormandi) [inline bitrise/bitrise.go:66]: Purely out of curiosity... what's the reason behind this change? Is there a difference between the two ways in Go?

Comment (gaborszakacs) [reply bitrise/bitrise.go:66]: No, this is the very same. Probably I did something with `cl` before returning it at some point then that got deleted (?), not sure. This change can be reversed.

**Comment by zormandi** [comment]
I'm not sure I have enough context for a review, but it LGTM.

**Review by zormandi — DISMISSED** [review]

File: featureflag/feature_flag.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+// TODO: find out where to add graceful teardown to Buffalo
+func Close() error {
+	if Client == nil {
+		return nil
+	}
+
+	return Client.Close()
+}
+
+func initOfflineClient() error {
```
Comment (zormandi) [inline featureflag/feature_flag.go:60]: Great job thinking about the offline client! This should be the "standard" way for us to integrate LaunchDarkly. 👍

File: featureflag/feature_flag.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	flagValue, err := Client.BoolVariation(flagKey, user, fallback)
+	if err != nil {
+		return fallback
+	}
+
+	return flagValue
+}
+
+// Close ...
+// TODO: find out where to add graceful teardown to Buffalo
```
Comment (zormandi) [inline featureflag/feature_flag.go:51]: I think we should do this now rather than leave it for later. It feels like we're creating tech debt. I mean we can push this into production to see it working but let's find a place for this today.

**Review by zormandi — APPROVED** [review]
